.github/workflows/buildtest.yml

```diff
@@ -28,37 +28,29 @@ jobs:
     defaults:
       run:
         shell: bash
-    env:
-      CMAKE_PRESET: docker
-      BUILD_TARGETS: full
     steps:
       - run: echo ${{ github.repository }}
       # Due to a bug in runner action the variables $GITHUB_WORKSPACE and ${{ github.workspace }} are different inside a container. https://github.com/actions/runner/issues/2058
       # The repo gets cloned to `/__w/4C/4C` ($GITHUB_WORKSPACE) while ${{ github.workspace }} points to `/home/runner/work/4C/4C`.`
       # Use $GITHUB_WORKSPACE instead of ${{ github.workspace }}
       - uses: actions/checkout@v4
-      - name: Fix git
-        run: git config --global --add safe.directory $GITHUB_WORKSPACE
       - name: info
         run: |
           pwd; ls -l
-      - name: Build
-        run: |
-          echo ${{ github.workspace }}
-          echo $GITHUB_WORKSPACE
-          cmake --version; echo ${{ env.CMAKE_PRESET }}
-          mkdir -p $GITHUB_WORKSPACE/../build
-          cd $GITHUB_WORKSPACE
-          ./utilities/set_up_dev_env.sh
-          cd $GITHUB_WORKSPACE/../build
-          cmake $GITHUB_WORKSPACE --fresh --preset=${{ env.CMAKE_PRESET }}
-          echo Building the following targets ${{ env.BUILD_TARGETS }}
-          time cmake --build . --target ${{ env.BUILD_TARGETS }} -- -j `nproc` 2>&1
+      - uses: ./.github/actions/configure_4C
+        with:
+          cmake-preset: docker
+          build-directory: ${{ github.workspace }}/../build
```

Comment:
I'm a bit confused whats happening here (not your code. But what the github runner is doing). You set the build directory to `${{ github.workspace }}/../build` which according to the [log](https://github.com/4C-multiphysics/4C/actions/runs/11763968417/job/32768579260#step:6:27) is `BUILD_DIRECTORY: /home/runner/work/4C/4C/../build`. At the end of the configuring the [log](https://github.com/4C-multiphysics/4C/actions/runs/11763968417/job/32768579260#step:6:522) says `Build files have been written to: /__w/4C/build`. The subsequent build works though. So it seems that`$GITHUB_WORKSPACE` and `${{ github.workspace }}` are after all the same directory.

Reply:
Yes, that also confused me.

Reply:
Slightly related: how about using a subfolder `build` inside the repo in the future? Actually, I think we should even ignore this folder in the `.gitignore` since this is how I nowadays recommend setting up a build.

Reply:
good idea. I do that for a few other jobs to avoid unnecessary copying and avoiding relative links for artifacts. Only for clang-tidy, it might not work (or we might need to exclude the build-folder then)

Reply:
Why should it not work for clang-tidy? We are normally using the `-p` arg which tells clang-tidy where to look for compile command lines. I use that locally all the time.

Reply:
We then also get warnings/errors from our dependencies (ArborX, Google test, etc).

Reply:
Ah I see, it is the combination of `-p compile_commands.json` and the additional working directory. I am fine with leaving it as is.